HIP-1.md

```diff
@@ -0,0 +1,146 @@
+| hip | title               | status | author | created    |
+| --- | ------------------- | ------ | ------ | ---------- |
+| 1   | Sovereign consensus | Draft  | asaj   | 2022-10-04 |
+
+### **Brief Summary / Abstract**
+
+Defines the specification for sovereign consensus, which puts control of the interchain security model in the hands of message recipients.
+
+### **Motivation**
+
+Interchain security models offer different tradeoffs with respect to security, latency, and cost. For example:
+
+1. Multisig models prioritize latency and cost over security
+2. Bonding models (e.g. proof-of-stake) prioritize latency and security over cost
+3. Optimistic models prioritize security and cost over latency
+4. Natively verified models prioritize security over cost (and sometimes latency)
+
+Interchain applications necessarily adopt the tradeoffs of their interchain security models. Rather than offering a one-size-fits-all product, sovereign consensus allows applications to choose (or construct) a security model that offers the best choice of tradeoffs for that application.
+
+Furthermore, the modularity of sovereign consensus allows Hyperlane to be forward compatible with new security models as they emerge.
+
+### **Tech Spec**
+
+Sovereign consensus is powered by `Interchain Security Modules` or `ISMs`, which specify the rules by which a `recipient` will accept an interchain message.
+
+Message `recipients` should specify their `ISM` via the `interchainSecurityModule()` view function.
+
+The `Inbox` contract, when delivering a message via `Inbox.process()`, calls `recipient.interchainSecurityModule().accept()`, which accepts or rejects the message.
+Relayers can inject arbitrary off-chain data (e.g. validator signatures, zero-knowledge proofs, etc.) into `accept()`, allowing for a wide degree of flexibility in `ISM` design.
+
+Some possible ISMs:
+
+- Multisig: Given a message, specifies a set of signers and a threshold that need to have signed `_root` in order to accept a message. Note that the set and threshold can vary based on message content.
+- Optimistic: Given a message, specifies a set of watchers that can pause the `ISM`. Otherwise, merkle roots are accepted, and after a fraud window has expired messages can be proved against these roots.
+- ZKP: Verifies a zero-knowledge-proof of the origin chain's light client protocol, with a merkle proof of the corresponding Hyperlane Outbox's merkle root.
```

Comment:
would it make more sense to use the global origin state root instead of the Outbox for broader compatibility?

Reply:
in these constructions, it would make sense to amortize acceptance criteria across messages. Is that spec compatible?

Reply:
> would it make more sense to use the global origin state root instead of the Outbox for broader compatibility?

The ZKP probably proves the state root, and then we prove the Outbox root, and then the message, with two additional merkle proofs.

Reply:
> in these constructions, it would make sense to amortize acceptance criteria across messages. Is that spec compatible?

Yeah you can cache the global state root and/or the Outbox root